jni/include/faiss_wrapper.h

```diff
@@ -33,6 +33,18 @@ namespace knn_jni {
         // Return a pointer to the loaded index
         jlong LoadIndex(knn_jni::JNIUtilInterface * jniUtil, JNIEnv * env, jstring indexPathJ);
 
+        // Check if a loaded index is an IVFPQ index with l2 space type
+        bool IsIndexIVFPQL2(jlong indexPointerJ);
```

Comment:
should we rename this to isSharedStateRequired?

Reply:
went back and forth on this a little with @heemin32. This change assumes that the shared state will only work for IVFPQ-l2. Specifically, for the case of freeing shared state, we do not indicate what type of index the shared state belongs to, so it cannot be extended with the interface now to support other types of shared state. For this reason I thought it might make sense to be explicit about what is supported and then generalize in 2.14 or 2.15.

However, for future readability, I do think isSharedStateRequired might be better. Ill make this change.